kasa/device.py

```diff
@@ -152,6 +152,22 @@ class WifiNetwork:
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class DeviceModelInfo:
+    """Device Model Information."""
+
+    short_name: str
+    long_name: str
+    brand: str
+    device_family: str
+    device_type: DeviceType
+    hardware_version: str
+    firmare_version: str
+    firmware_build: str
```

Comment:
I'm not sure if we want to go that detailed here?

How do you envison short_name/long_name/brand to be used for non-smartcameras?

Reply:
Actually it's the `smart` devices I'm thinking of with `short_name` `long_name`. The model in the device info is shorter than in the discovery info, e.g. `L530` vs `L530E`.

In terms of the detail I'd like to replace the duplicated device specific logic that is sitting in `dump_devinfo` and `generate_supported` with this same implementation in `iot` and `smart` devices. The detail does not **need** to be exposed as part of the public api, that's up for discussion and I don't feel strongly.

Reply:
Sounds good to me 👍